### DISCUSSION TITLE
False available updates?

### DISCUSSION THREAD

**Discussion by whipster5** [uyuni-project/uyuni#9484] [Q&A]
Hello everyone,

Could you help me figure out what I might be doing wrong? I have an Uyuni server (2024.08) and a Rocky Linux 8 client. After fully updating and rebooting my client, it still shows missing updates:

```
spacecmd {SSM:0}> system_listupgrades myclient
perl-IO-Socket-SSL-2.066-4.module+el8.9.0+1517+11a26ffb.noarch
perl-Net-SSLeay-1.88-2.module+el8.9.0+1517+11a26ffb.x86_64

```
However, when I run yum check-update on the client, it shows no updates. These Perl modules and versions are associated with Perl 5.24, but my client has Perl 5.26 installed. I am also using the Rocky Linux AppStream repository.

Am I missing something, or is this some kind of strange behavior?

Thank you in advance for your help!

**Comment by RenownTwiz** [comment]
I'm certainly not the expert here so take my feedback with a bit of caution.

We run a lot of Oracle Enterprise Linux, which is very similar to Rocky.  For the longest time, we had the same/similar issue you see with SuseManager and Uyuni.  I'm not sure which Uyuni version the fixes went in for correctly interpreting Appstreams, it might have been 2024.10?  So you may just need to update.  This is fixed in the latest Uyuni releases with maybe an exception that I'll explain here in a minute.  If you look at what packages it's complaining about, I bet that the packages are all from modules in the Appstreams, and Uyuni is showing packages from "newer" modules than the ones your server is currently set to use.

The exception that I spoke about has to do with how you promote packages through your channels.  We prior to 2024.10 had some python scripts that used the APIs:

client.channel.software.mergePackages
client.channel.software.mergeErrata

To move all the packages through our own lifecycle / workflow from distro channels --> nonprod channels --> production channels.

Even in the latest 2024.10 release, using the merge APIs does not seem to work quite right with appstreams.  I'm assuming some metadata related to the appstreams is getting lost and not propagated to the nonprod/production channels.  I've switched to using "Content Lifecycle" projects and it seems to work just fine doing it the new way.

When using the merge APIs I still see the same thing you see, when I look at a specific client, it shows packages as being out of date, plus if you look at the Software-->AppStreams tab it doesn't show what modules the server has selected.  It looks like the attached image.


![Software-AppStreams](https://github.com/user-attachments/assets/53101580-11c5-4fd0-81bf-9310ceff62d7)

Hope this maybe helps you.
